google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -85,6 +85,32 @@ void ToJson(std::chrono::system_clock::time_point const& field,
   j[name] = std::to_string(m);
 }
 
+nlohmann::json RemoveJsonKeysAndEmptyFields(std::string const& json_payload,
+                                            std::vector<std::string> keys) {
+  nlohmann::json::parser_callback_t remove_empty_call_back =
+      [&keys](int depth, nlohmann::json::parse_event_t event,
+              nlohmann::json& parsed) {
+        // Have this redundant check due to compile error that
+        // depth parameter is not being used.
+        if (depth < 0) return false;
```

Comment:
nit: Can you say `int /* depth */, ...` in the signature?

If not, I'd change this line to `(void)depth;`